### PR TITLE
Added ability to hide edit locks and groups in editor view

### LINKED ISSUE #17648: Ability to hide crosses, locks, and 'make children unselectable' icons from editor view
This is a feature request to add a view option (or multiple options) to hide the display of crosses, locks, and 'make children unselectable' icons from the editor view.

The reason is in more complex scenes, the amount of locks, unselectable children icons and crosses that appear become extremely distracting and clutter up the editor view too much. 

'Locking' and making children unselectable are so you can ignore those nodes when clicking/selecting other nodes. However as it currently is, locking/making children unselectable means more icons that actually distract instead. So actually I think it would be better to default to not showing the locks/unselectable icons in the editor view. Other graphics tools that I can think of (Photoshop, Illustrator, Sketchup) that have the ability to lock layers/objects also don't show 'locked' icons in the editing view.

The option(s) could maybe be in the View menu dropdown, or in the Editor Settings.

### PR DESCRIPTION
**Pull request by Piet-G**
Added ability to hide edit lock and groups in editor view.

This fixes #17648

### CODE REVIEW

**Comment by groud** [comment]
Good job, but I would make this a single option. There no need to split this into two options, one is enough.

**Comment by DualMatrix** [comment]
You're right, combined the two options into one.

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -4543,6 +4556,8 @@ CanvasItemEditor::CanvasItemEditor(EditorNode *p_editor) {
 	p->add_check_shortcut(ED_SHORTCUT("canvas_item_editor/show_guides", TTR("Show Guides"), KEY_Y), SHOW_GUIDES);
 	p->add_check_shortcut(ED_SHORTCUT("canvas_item_editor/show_origin", TTR("Show Origin")), SHOW_ORIGIN);
 	p->add_check_shortcut(ED_SHORTCUT("canvas_item_editor/show_viewport", TTR("Show Viewport")), SHOW_VIEWPORT);
+	p->add_check_shortcut(ED_SHORTCUT("canvas_item_editor/show_edit_locks", TTR("Show Edit Locks")), SHOW_EDIT_LOCKS);
```
Comment (groud) [inline editor/plugins/canvas_item_editor_plugin.cpp]: The text entry should be made clearer. Something like "Show groups and locks icons".

**Comment by DualMatrix** [comment]
Changed text entry to "Show Group And Lock Icons"

**Review by groud — APPROVED** [review]

**Comment by groud** [comment]
Awesome, thanks ! :)

**Pull request merged**